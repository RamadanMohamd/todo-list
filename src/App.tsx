import { useState } from "react";
import { Button } from "@/components/ui/button";


function App() {
  const [count, setCount] = useState(0);

  return (
    <div className="flex items-center justify-center flex-col min-h-screen py-2 bg-gray-100">
      <h1>Vite + React</h1>
        <Button variant="default" className="my-4" onClick={() => setCount((count) => count + 1)}>count is {count}</Button>
        <p>
          Edit <code>src/App.tsx</code> and save to test HMR
        </p>
      <p className="read-the-docs">Click on the Vite and React logos to learn more</p>
    </div>
  );
}

export default App;
