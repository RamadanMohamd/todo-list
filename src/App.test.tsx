import '@testing-library/jest-dom';
import { render, screen, fireEvent } from "@testing-library/react";
import App from "./App";

describe("App Component", () => {
  test("renders Vite and React logos", () => {
    render(<App />);
    const viteLogo = screen.getByAltText(/vite logo/i);
    const reactLogo = screen.getByAltText(/react logo/i);
    expect(viteLogo).toBeInTheDocument();
    expect(reactLogo).toBeInTheDocument();
  });

  test("renders heading", () => {
    render(<App />);
    const heading = screen.getByRole("heading", { name: /vite \+ react/i });
    expect(heading).toBeInTheDocument();
  });

  test("renders button with initial count", () => {
    render(<App />);
    const button = screen.getByRole("button", { name: /count is 0/i });
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent("count is 0");
  });

  test("renders paragraph with read the docs instruction", () => {
    render(<App />);
    const readTheDocs = screen.getByText(/click on the vite and react logos to learn more/i);
    expect(readTheDocs).toBeInTheDocument();
  });

  test("increments count when button is clicked", () => {
    render(<App />);

    const button = screen.getByRole("button", { name: /count is 0/i });
    expect(button).toHaveTextContent("count is 0");
    fireEvent.click(button);
    expect(button).toHaveTextContent("count is 1");
    fireEvent.click(button);
    expect(button).toHaveTextContent("count is 2");
  });
});