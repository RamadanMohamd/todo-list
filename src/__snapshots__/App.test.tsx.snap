// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<DocumentFragment>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <button>
      count is 0
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</DocumentFragment>
`;
